### ISSUE TITLE
Request: add option to hide the applet when no battery devices available

### ISSUE THREAD

**Issue by mrpeabody** [codilia/upower-battery#20]
Thank you the excellent extension, it looks and works really well! 

I'm wondering if you could consider adding an option to completely hide the applet in the panel when the battery devices are not available. Currently when I turn off my Bluetooth headset I see a gap and small ellipsis when hover over it that shows nothing.

Thank you very much for your great work!

**Comment by mrpeabody** [comment]
I've made a patch (attached) and tested -- seems to be working really well.
[extension.zip](https://github.com/codilia/upower-battery/files/15461744/extension.zip)


```
diff --git a/extension.js b/extension.js
index 8d43e64..2fb7a0d 100644
--- a/extension.js
+++ b/extension.js
@@ -77,6 +77,23 @@ export default class UpowerBatteryExtension {
 
 	_update() {
 		const devices = this._findDevices();
+
+		// Hide the inidicator if there are no devices to display
+		if (devices.length === 0) {
+			if (this._indicator) {
+				this._indicator.destroy();
+				this._indicator = null;
+			}
+			return;
+		}
+
+		// we got devices but no indiator yet, let's create one
+		if (!this._indicator) {
+			this._indicator = new Indicator.Indicator();
+			this._proxies = {};
+			Main.panel.addToStatusArea('upowerBattery', this._indicator, 3, 'right');
+		}
+
 		this._indicator.refresh(devices);
 	} 

```

**Comment by ppiastucki** [comment]
Would you please try this, a bit simpler, fix: https://github.com/codilia/upower-battery/pull/21?

**Comment by mrpeabody** [comment]
> Would you please try this, a bit simpler, fix: #21?

Works like a charm! Nice fix, I'm not super familiar with the Gnome JS API :+1:

**Comment by ppiastucki** [comment]
Thank you!
